lms/djangoapps/teams/csv.py

```diff
@@ -19,3 +27,199 @@ def load_team_membership_csv(course, response):
         "Team membership CSV download is not yet implemented."
     )
     response.write(not_implemented_message + "\n")
+
+
+class TeamMembershipImportManager(object):
+    """
+    A manager class that is responsible the import process of csv file including validation and creation of
+    team_courseteam and teams_courseteammembership objects.
+    """
+
+    def __init__(self, course):
+        self.validation_errors = []
+        self.teamset_ids = []
+        self.user_ids_by_teamset_id = {}
+        self.teamset_ids = []
+        self.number_of_record_added = 0
+        # stores the course module that will be used to get course metadata
+        self.course_module = ''
+        # stores the course for which we are populating teams
+        self.course = course
+        self.max_errors = 0
+
+    @property
+    def import_succeeded(self):
+        """
+        Helper wrapper that tells us the status of the import
+        """
+        return not self.validation_errors
+
+    def set_team_membership_from_csv(self, input_file):
+        """
+        Assigns team membership based on the content of an uploaded CSV file.
+        Returns true if there were no issues.
+        """
+        self.course_module = modulestore().get_course(self.course.id)
+        reader = csv.DictReader((line.decode('utf-8-sig').strip() for line in input_file.readlines()))
+        self.teamset_ids = reader.fieldnames[2:]
+        row_dictionaries = []
+        if self.validate_teamsets():
+            # process student rows:
+            for row in reader:
+                username = row['user']
+                if not username:
+                    continue
+                user = self.get_user(username)
+                if user is None:
+                    continue
+                if not self.validate_user_enrolled_in_course(user):
+                    row['user'] = None
+                    continue
+                row['user'] = user
+
+                if not self.validate_user_assignment_to_team_and_teamset(row):
+                    return False
+                row_dictionaries.append(row)
+
+            if not self.validation_errors:
+                for row in row_dictionaries:
+                    self.add_user_to_team(row)
+                return True
+            else:
+                return False
+        return False
+
+    def validate_teamsets(self):
```

Comment:
Nit: this method does two things, which isn't really made clear until farther down in the docstring.  How about a separate method called `load_user_ids_by_teamset_id()` that you can call after `validate_teamset()` that populates the `user_ids_by_teamset_id` field?